cmd/kube-controller-manager/app/options/tenantcontroller.go

```diff
@@ -35,6 +35,7 @@ func (o *TenantControllerOptions) AddFlags(fs *pflag.FlagSet) {
 
 	fs.DurationVar(&o.TenantSyncPeriod.Duration, "tenant-sync-period", o.TenantSyncPeriod.Duration, "The period for syncing tenant life-cycle updates")
 	fs.Int32Var(&o.ConcurrentTenantSyncs, "concurrent-tenant-syncs", o.ConcurrentTenantSyncs, "The number of tenant objects that are allowed to sync concurrently. Larger number = more responsive tenant termination, but more CPU (and network) load")
+	fs.StringVar(&o.DefaultNetworkTemplatePath, "default-network-template-path", o.DefaultNetworkTemplatePath, "The path to the template file for the default network spec in JSON of tenants. Empty means system won't create  the default network for new tenants")
```

Comment:
Will it be a problem if there is no default network object?

Reply:
If a tenant has no default network object, it will have some problems with DNS & kubernetes service. We don't want to enforce every tenant __MUST__ have default network rule, as tenant admin can have choose to work on non-default networks.

In flat network, if there exists multi tenancy, we still need different kube-dns for each tenant. In current design, dns service is per network, not per tenant.  We need explicit default network object (to provide dns service info). For the question of whether there is other options in flat network, I'll open an issue to track it.